doc/website/blog/2024-04-04-top-usability-issues.mdx

```diff
@@ -0,0 +1,233 @@
+above.
+
+But while technically correct, in terms of usability, they provide a very poor developer experience.
+
+Consider this directory structure from a typical Flyway project:
+
+```
+.
+├── V1__create_table1.sql
+├── V2__create_table_second.sql
+└── V3_1__add_comments.sql
+```
+
+To have a migration directory describe the current schema of a database is like describing your home by
+listing the needed steps to build it. It's correct, but it's not very useful.
+
+:::info Issue #1: Migrations as a source of truth
+
+**Problem**: As a human, I can't understand the current state of the database by looking at the migration files.
+I need to run them in order to understand the current state of the database.
+
+**Solution**: Use a declarative [schema-as-code approach](/blog/2024/01/24/why-schema-as-code) to describe the current state of the database.
+Then, in every version of our project, we can read the current schema in plain text and understand it.
+
+:::
+
+### Issue #2: Manual planning
+
+Classic migration tools typically require you to write the migration scripts manually. This process can be tedious
+and error-prone, especially if you have developers who don't have a lot of experience with operating databases.
+
+Classic migration tools were created a time when tech-stacks were much simpler and less diverse. In addition,
+many organizations employed a DBA who could serve as a technical authority for database changes. Nowadays, developers
+are expected to "own" their databases and DBAs are seriously outnumbered.  This means that developers are expected to
+write migration scripts, even if they don't have a lot of experience with databases.
+
+Doing something, when it's not in your area of expertise, especially when its critical and risky like database migrations,
+can be a daunting and stressful task.  Even if it is something you are good at, it's still something that requires
+your attention and focus.
+
+It's important to note that some ORMs (such as Django and Prisma) do stand out in their ability to generate migration
+scripts automatically (with some important limitations).  But for the most part, developers are expected to write
+migration scripts manually.
+
+:::info Issue #2: Manual planning
+
+**Problem**: Writing migration scripts can be stressful and error-prone, especially for
+developers who don't have a lot of experience with databases.  They need to first know what the
+current state of the database is, then they need to know what the desired state is, and then they need to write
+a script that will take the database from the current state to the desired state.  Some changes are trivial, but
+some require consideration and research.
+
+**Solution**: Use a tool that can [automatically generate migration](/versioned/diff) scripts for you.
+This way, you can focus on the desired state of the database and let the tool figure out how to get there.
+
+:::
+
+### Issue #3: Working in parallel
```

Comment:
Maybe you can link to [this](https://atlasgo.io/lint/analyzers#non-linear-changes) in the problem/solution section.